src/capi/typeobject.cpp

```diff
@@ -1022,14 +1023,13 @@ Box* slotTpGetattrHookInternal(Box* self, BoxedString* name, GetattrRewriteArgs*
                     } else
                         rtn->addGuard(0);
```

Comment:
I think we might need to make this `rewrite_args = NULL` instead, since it's not clearing a potential capi exception.  I'm surprised that the change you made here helped, since I would have thought these guards would cause the rewrite to abort anyway.